core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -605,6 +616,9 @@ class DynamicBrokerConfig(private val kafkaConfig: KafkaConfig) extends Logging
   }
 }
 
+/**
+ * Implement [[org.apache.kafka.server.config.BrokerReconfigurable]] instead.
+ */
 trait BrokerReconfigurable {
```

Comment:
@frankvicky Do you want to include replacing `kafka.server.BrokerReconfigurable` with `org.apache.kafka.server.config.BrokerReconfigurable` in this PR? Or you prefer to do it in next PR? Thanks.

Reply:
I don't plan to replace `kafka.server.BrokerReconfigurable` with `org.apache.kafka.server.config.BrokerReconfigurable` in this patch.
IMHO, we should do the replacing when we migrate these subclass of `kafka.server.BrokerReconfigurable`. In this way, we could keep minimum changes for each patch and eventually remove `kafka.server.BrokerReconfigurable`.